src/test/java/kitchenpos/application/TableServiceTest.java

```diff
@@ -0,0 +1,221 @@
+        orderTableRequest.setNumberOfGuests(0);
+        orderTableRequest.setEmpty(true);
+
+        // when
+        final OrderTable createdTable = orderTableDao.save(orderTableRequest);
+
+        // then
+        assertThat(createdTable.getId()).isNotNull();
+    }
+
+    @Test
+    @DisplayName("생성된 테이블 정보들을 가져온다.")
+    void test() {
+        // given
+
+        // when
+        final List<OrderTable> tables = tableService.list();
+
+        // then
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(tables).isNotEmpty();
+            final OrderTable createdTable = tables.get(tables.size() - 1);
+            softly.assertThat(createdTable.getId()).isEqualTo(emptyTable.getId());
+        });
+    }
+
+    @Nested
+    @DisplayName("테이블 상태 변경 테스트")
+    class ChangeEmptyStatusTest {
+
+        @Test
+        @DisplayName("상태변경에 성공한다. - empty true -> false")
+        void successFromTrueToFalse() {
+            // given
+            final OrderTable changeRequest = new OrderTable();
+            changeRequest.setEmpty(false);
+
+            // when
+            final OrderTable response = tableService.changeEmpty(emptyTable.getId(), changeRequest);
+
+            // then
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(response.getId()).isEqualTo(emptyTable.getId());
+                softly.assertThat(response.isEmpty()).isFalse();
+            });
+        }
+
+        @Test
+        @DisplayName("상태변경에 성공한다. - empty true -> false")
+        void successFromFalseToTrue() {
+            // given
+            final OrderTable changeRequest = new OrderTable();
+            changeRequest.setEmpty(true);
+
+            // when
+            final OrderTable response = tableService.changeEmpty(notEmptyTable.getId(), changeRequest);
+
+            // then
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(response.getId()).isEqualTo(notEmptyTable.getId());
+                softly.assertThat(response.isEmpty()).isTrue();
+            });
+        }
+
+        @Test
+        @DisplayName("테이블 그룹에 속해있으면 상태를 변경할 수 없다.")
+        void throwExceptionWithGroupedTable() {
+            // given
+            final TableGroup tableGroup = new TableGroup();
+            tableGroup.setCreatedDate(LocalDateTime.now());
+            final TableGroup createdTableGroup = tableGroupDao.save(tableGroup);
+
+            notEmptyTable.setId(null);
+            notEmptyTable.setTableGroupId(createdTableGroup.getId());
+            notEmptyTable = orderTableDao.save(notEmptyTable);
+
+            final OrderTable changeRequest = new OrderTable();
+            changeRequest.setEmpty(true);
+
+            // when
+            // then
+            Assertions.assertThatThrownBy(() -> tableService.changeEmpty(notEmptyTable.getId(), changeRequest))
+                    .isInstanceOf(IllegalArgumentException.class);
```

Comment:
> Only one method invocation is expected when testing runtime exceptions
> 
> When verifying that code raises a runtime exception, a good practice is to avoid having multiple method calls inside the tested code, to be explicit about which method call is expected to raise the exception.
It increases the clarity of the test, and avoid incorrect testing when another method is actually raising the exception.

런타임 예외가 발생하는지 확인할 때, 어느 부분에서 예외가 발생하는지 명확하게 하기 위해서 여러 메서드 호출을 포함하지 않도록 하는 게 좋다고 합니다!

위 코드에서는 `changeEmpty` 메서드에서 예외가 발생하는지 확인하기 위한 테스트니까 `notEmptyTable.getId()`를 변수로 분리하는 방법을 사용할 수 있겠네요.

Reply:
와 이부분은 생각 못했던 부분인데, 확실히 예외테스트시 여러 메서드가 호출이 되면 명확한 문제시점을 확정하지 못하겠네요...!

위와같은 방식으로 되어있던 부분들 모두 변경했습니다!